src/search/query-mode.ts

```diff
@@ -0,0 +1,57 @@
+export enum QueryMode {
+  RAW = 'raw',
+  SMART = 'smart',
+  DEPRECATED = 'deprecated',
+}
+
+export function queryVariants(query: string, type: QueryMode) {
+  switch (type) {
+    case QueryMode.RAW:
+      return new Map([
+        ['?query', query],
+        ['?booleanQuery', query],
+      ]);
+    case QueryMode.SMART:
+      return new Map([
+        ['?query', stringQuery(query)],
+        ['?booleanQuery', booleanQuery(stringQuery(query))],
+      ]);
+    case QueryMode.DEPRECATED:
+      return new Map([['?query', query]]);
+  }
+}
+
+const stringQuery = (query: string) =>
+  query.toLowerCase().replace(/\s+/g, ' ').trim();
+
+const booleanQuery = (query: string) =>
```

Comment:
A late thought: or should we rename `booleanQuery` to `virtuosoQuery`, to make very clear in which situation it can be used (and to prevent that the same variable is going to be used in a query for another triplestore, where it could yield unexpected results, e.g. because the `escape` and `filterStopWords` functions currently are Virtuoso-specific)?

Reply:
Agreed; changed to `virtuosoQuery`. I originally had `booleanQuery` to sort of hide the fact that we’re optimising for a specific triple store, but yeah, after all that’s really what we’re doing.